packages/appeals-service-api/src/lib/notify.js

```diff
@@ -0,0 +1,69 @@
+const { NotifyClient } = require('notifications-node-client');
+const fetch = require('node-fetch');
+const config = require('./config');
+const logger = require('./logger');
+const { getLpa } = require('../services/lpa.service');
+
+function getAddress(siteAddress) {
+  let address = '';
+  if (siteAddress.addressLine1) {
+    address = `${siteAddress.addressLine1}\n`;
+  }
+  if (siteAddress.addressLine2) {
+    address = `${address}${siteAddress.addressLine2}\n`;
+  }
+  if (siteAddress.town) {
+    address = `${address}${siteAddress.town}\n`;
+  }
+  if (siteAddress.county) {
+    address = `${address}${siteAddress.county}\n`;
+  }
+  if (siteAddress.postcode) {
+    address = `${address}${siteAddress.postcode}`;
+  }
+  return address;
+}
+
+async function sendEmail(appeal) {
+  logger.info(`Sending email...`);
+  logger.debug(`appeal = \n${JSON.stringify(appeal, null, 2)}`);
+  const { baseUrl, serviceId, apiKey, templateId } = config.services.notify;
+  const args = [];
+  if (baseUrl) {
+    args.push(baseUrl, serviceId);
+  }
+  args.push(apiKey);
+  const notifyClient = new NotifyClient(...args);
+  logger.debug(`Notify client = \n${JSON.stringify(notifyClient.apiClient, null, 2)}`);
+  const applicationId = appeal.id;
+  const documentId = appeal.appealSubmission.appealPDFStatement.uploadedFile.id;
+  const url = `${config.documents.url}/api/v1/${applicationId}/${documentId}`;
+  let pdfFile;
+  try {
+    const respFile = await fetch(`${url}/file`);
+    const arrayBuffer = await respFile.arrayBuffer();
+    pdfFile = Buffer.from(arrayBuffer);
+    const lpa = await getLpa(appeal.lpaCode);
+    const options = {
+      personalisation: {
+        'appeal site address': getAddress(appeal.appealSiteSection.siteAddress),
+        'link to appeal submission pdf': notifyClient.prepareUpload(pdfFile),
+        'local planning department': lpa.name,
+        name: appeal.aboutYouSection.yourDetails.name,
+      },
+      reference: appeal.id,
+    };
+    logger.debug(`templateId = ${templateId}`);
+    const destinationEmail = appeal.aboutYouSection.yourDetails.email;
+    logger.debug(`destinationEmail = ${destinationEmail}`);
+    logger.debug(`options = \n${JSON.stringify(options, null, 2)}`);
+    await notifyClient.sendEmail(templateId, destinationEmail, options);
+  } catch (err) {
+    const msg = 'Problem sending email';
+    logger.error({ err }, msg);
+  }
+}
+
+module.exports = {
+  sendEmail,
```

Comment:
I'd strongly advocate for a unit test for this file.

Reply:
In progress